test/src/test/java/jenkins/model/JenkinsTest.java

```diff
@@ -471,4 +480,116 @@ public void onOnline(Computer c, TaskListener listener) throws IOException, Inte
             throw new IOException("Something happened (the listener always throws this exception)");
         }
     }
+    
+    @Test
+    @Issue("JENKINS-39465")
+    public void agentProtocols_singleEnable_roundtrip() throws Exception {
+        final Set<String> defaultProtocols = Collections.unmodifiableSet(j.jenkins.getAgentProtocols());
+        Assume.assumeThat("We assume that JNLP3-connect is disabled", 
+                defaultProtocols, not(hasItem("JNLP3-connect")));
```

Comment:
Is there a reason you are not just using `assertEquals` to specify the exact set of protocols? So that the test does not need to be updated if and when new protocols are added? But that seems like a trivial amount of work compared to the work of actually integrating the new protocol, and compared to the benefit of having a test which is both stronger (catches more mistakes) and simpler (easier to read).